File: C#/UDEMY_CSharp_Completo_POO/_078_ExerciciosFixacaoListas/_078_ExerciciosFixacaoListas/Employee.cs
﻿using System.Globalization;

namespace _078_ExerciciosFixacaoListas
{
    class Employee
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Salary { get; private set; }

        public Employee(int id, string name, double salary)
        {
            Id = id;
            Name = name;
            Salary = salary;
        }

        public void IncreaseSalary(double percentage)
        {
            Salary += Salary * percentage / 100.0;
        }

        public override string ToString()
        {
            return Id
            + ", "
            + Name
            + ", U$ "
            + Salary.ToString("F2", CultureInfo.InvariantCulture);
        }
    }
}

File: estudosJava/CursoJavaCompletoUdemy/ExerciciosFixacao/Ex002/src/main/java/entities/Employee.java
package entities;

public class Employee {
    public String name;
    public double grossSalary;
    public double tax;

    public double netSalary(){
        return grossSalary - tax;
    }
    public void increaseSalary(double percentage) {
        double increase = grossSalary * percentage / 100;
        grossSalary = grossSalary + increase;
    }

    @Override
    public String toString() {
        return name + ", $ "
                + String.format("%.2f", netSalary());
    }
}
